Automattic/wp-calypso
# Plans 2023: Remove loading-placeholder component and use from @automattic/components

Fixes https://github.com/Automattic/wp-calypso/issues/79008

## Proposed Changes

- Removes the custom loading-placeholder component and imports the one from `@automattic/components`. This needs to be the way going forward for similar cases (where existing components exists, let's seek reuse / and customise if necessary)
- Some customisations to the existing component to be able to tailor it for the one case in the sub-header component
    - the other cases look fine with the default styling
    - not sure if the use case in sub-header needs all that custom styling (different height and border-radius). In general, if we don't need to customise too much these things the better (for cohesion and keep code/complexity minimum)

### Media 

<img width="400" alt="Screenshot 2023-11-02 at 2 53 18 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/e89ac233-6862-402c-8ccb-9aefec42462a">

## Testing Instructions



* There are many uses of the loading-placeholder from `@automattic/components`, although I don't think we need to test more than what we have under `plans-features-main` and `plans-grid`
* Go to `/start/onboarding-pm`, pick a paid domain and head to plans. it should show the placeholder as in media above
    * May need throttling on network to delay the processing or force the condition to true in the code changes
* Other cases appear in tooltip over "custom domain" feature in the plan table and the modal when selecting "paid domain" and the "free plan"

## Pre-merge Checklist
